main.go

```diff
@@ -95,10 +95,17 @@ func marshalAndWrite(msg proto.Message, c *net.UDPConn, dst *net.UDPAddr, kv llo
 
 func readAndUnmarshal(c *net.UDPConn, kv llog.KV) (*lproto.TxMsg, *net.UDPAddr, bool) {
 	b := make([]byte, 1024)
+	c.SetReadDeadline(time.Now().Add(time.Second * 5))
 	n, addr, err := c.ReadFromUDP(b)
 	if err != nil {
-		kv["err"] = err
-		llog.Error("error reading from udp socket", kv)
+		if nerr, ok := err.(net.Error); ok && nerr.Timeout() {
+			//the master will not respond after the transaction is complete
+			//so this is expected that it'll timeout for each transaction
+			llog.Info("timed out reading from udp socket", kv)
```

Comment:
Actually, this function is used on the master's side, and the slave already logs elsewhere that the transaction ended. Maybe better to not log here, or make it debug.